src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -38,6 +37,32 @@ public static AnalyzeResult<RemoteResolveResult> Analyze(this GraphNode<RemoteRe
             List<DowngradeResult<RemoteResolveResult>> downgrades,
             List<GraphNode<RemoteResolveResult>> cycles)
         {
+            var workingDowngrades = RentDowngradesDictionary();
+
+            root.ForEach((node, context) => WalkTreeCheckCycleAndNearestWins(context, node), CreateState(cycles, workingDowngrades));
+
+#if NET45
```

Comment:
Good to know, yes that works. I think next time the TFM changes all of these if defs would need to be checked manually, but using `IS_DESKTOP` does make this easier.